src/main/java/it/gov/pagopa/apiconfig/core/service/ChannelsService.java

```diff
@@ -112,13 +97,18 @@ public ChannelDetails createChannel(ChannelDetails channelDetails) {
   }
 
   public ChannelDetails updateChannel(String channelCode, ChannelDetails channelDetails) {
-    Long objId = getCanaliIfExists(channelCode).getId();
+    Canali canali = getCanaliIfExists(channelCode);
+    CanaliNodo canaliNodo = canali.getFkCanaliNodo();
 
     // add info for model mapper
     setInfoMapper(channelDetails);
 
-    var entity = modelMapper.map(channelDetails, Canali.class).toBuilder().id(objId).build();
-    canaliRepository.save(entity);
+    var newCanali = modelMapper.map(channelDetails, Canali.class).toBuilder()
+            .id(canali.getId())
+            .build();
+    var newCanaliNodo = newCanali.getFkCanaliNodo();
+    newCanaliNodo.setId(canaliNodo.getId());
+    canaliRepository.save(newCanali);
     return channelDetails;
```

Comment:
Why does it return the input and not the updated element?

Reply:
The model chosen for the output is the same as the passed input, for this reason it was decided to return that value. In a sense the return is the updated element, because is the representation via the model of the element updated on the db